### PR TITLE
chore: move plugins folder outside of Composer

### PR DESCRIPTION
**Pull request by a-b-r-o-w-n**
## Description

Also renames it to extensions.

This will help avoid yarn workspace issues with transient dependencies. Extensions should be considered external/isolated modules.

## Task Item

#minor

### CODE REVIEW

**Comment by a-b-r-o-w-n** [comment]
Can someone on windows test this with a clean checkout?

**Comment by benbrown** [comment]
This scares me! There is a big PR involving the azurePublish plugin.  Any chance we can let that land before we move all the files around?

**Comment by a-b-r-o-w-n** [comment]
@benbrown git should be able to handle the conflict since no source code was changed.

**Comment by a-b-r-o-w-n** [comment]
@benbrown if you tell me the branch you are working on I can do a merge locally to see.

**Comment by benbrown** [comment]
`wenyluo/provision`

File: Composer/packages/extension/src/extensionContext.ts
Comment (tonyanziano) [inline Composer/packages/extension/src/extensionContext.ts:107]: Is there anything using this that would benefit from knowing that this is deprecated? If it's only used internally can we just remove it?

File: Composer/scripts/compileExtensions.js
Comment (tonyanziano) [inline Composer/scripts/compileExtensions.js:74]: Just an idea: Would it be worth it to console.warn() here that if you want your extension built you need to define a `build` script in the `package.json`?

**Review by tonyanziano — COMMENTED** [review]
Some minor comments. I'll defer to how the merging of the azure publish branch goes

File: Composer/scripts/compileExtensions.js
Comment (a-b-r-o-w-n) [reply Composer/scripts/compileExtensions.js:74]: Good idea.

File: Composer/packages/extension/src/extensionContext.ts
Comment (a-b-r-o-w-n) [reply Composer/packages/extension/src/extensionContext.ts:107]: It's used in a single test file. I'll clean it up another time.

File: Composer/scripts/compileExtensions.js
Comment (tonyanziano) [inline Composer/scripts/compileExtensions.js:84]: 👍 👍 👍

**Comment by a-b-r-o-w-n** [comment]
That branch isn't set to merge until later next week. So we'll merge this when ready and I will manage the conflicts.

**Review by tonyanziano — APPROVED** [review]
Looks good once merge conflicts are fixed

**Review by tonyanziano — APPROVED** [review]
🚢

**Pull request merged**
